### PR TITLE
add tecMPT_ISSUANCE_NOT_FOUND code

### PR DESCRIPTION
**Pull request by shawnxie999**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/test/app/MPToken_test.cpp [hunk 1]
```diff
@@ -980,6 +980,24 @@ class MPToken_test : public beast::unit_test::suite
...
+
+        // Issuer fails trying to send fund after issuance was destroyed
+        {
+            Env env{*this, features};
+
+            MPTTester mptAlice(env, alice, {.holders = {&bob}});
+
+            mptAlice.create({.ownerCount = 1, .holderCount = 0});
+
+            mptAlice.authorize({.account = &bob});
```
Comment (gregtatcam) [inline src/test/app/MPToken_test.cpp:992]: Don't need to authorize.

File: src/xrpld/ledger/detail/View.cpp [hunk 2]
```diff
@@ -1875,7 +1875,7 @@ rippleCredit(
             view.update(sle);
         }
         else
-            return tecINTERNAL;
+            return tecMPT_ISSUANCE_NOT_FOUND;
```
Comment (gregtatcam) [inline src/xrpld/ledger/detail/View.cpp]: If a sender is not the issuer, need to check if mpt issuance doesn't exist. I.e., if it doesn't exist then the error is `tecMPT_ISSUANCE_NOT_FOUND` and if mpt token doesn't exist then the error is `tecNO_AUTH`.

Comment (shawnxie999) [reply src/xrpld/ledger/detail/View.cpp]: Could you clarify? The code currently does what you have described.

Comment (gregtatcam) [reply src/xrpld/ledger/detail/View.cpp]: If a sender is not the issuer then we are getting MPToken object. If this object doesn't exist then `tecNO_AUTH` is returned. This is not correct error if `MPTokenIssuance` object doesn't exist. I think we should check at the start of the function if `MPTokenIssuance` doesn't exist and return `tecMPT_ISSUANCE_NOT_FOUND`.

Comment (shawnxie999) [reply src/xrpld/ledger/detail/View.cpp]: i moved the check to the start of function

File: src/test/app/MPToken_test.cpp (hunk 1 above)
Comment (shawnxie999) [reply src/test/app/MPToken_test.cpp:992]: I think we should keep the test as it is, I added another test case where the holder didn't authorize

**Review by gregtatcam — APPROVED** [review]

**Pull request merged**